Example/Glia Sample Integration/Presenters/ContentViewPresenter.swift

```diff
@@ -5,86 +5,120 @@ import SwiftUI
-                self?.isAudioAvailable = queue?.isMediaTypeSupported(.audio) ?? false
-                self?.isChatAvailable = queue?.isMediaTypeSupported(.text) ?? false
-            }
-        }
-    }
-
-    // Configure listening of Glia's events
-    private func eventListener() {
-        Glia.sharedInstance.onEvent = { event in
-            switch event {
-            case .started:
-                break
-            case .engagementChanged:
-                break
-            case .ended:
-                break
-            case .minimized:
-                break
-            case .maximized:
-                break
+        // you need to initialize `CoreSDK` separately with your site's credentials.
+        initializeCoreSDK { [weak self] result in
+            switch result {
+            case .success:
+                self?.isSDKLoaded = true
+                self?.initializeAuthentication()
+                self?.checkQueueEngagementsAvailability()
+            case .failure(let error):
+                print("Error has happened: \(error.localizedDescription)")
             }
         }
     }
 
     // Start the engagement of a desired `EngagementKind`.
     func startEngagement(_ kind: EngagementKind) {
         do {
-            try Glia.sharedInstance.start(
-                kind,
-                configuration: gliaConfiguration.composeConfiguration(),
-                queueID: gliaConfiguration.queueId,
-                visitorContext: nil,
-                theme: theme
-            )
+            try Glia.sharedInstance.startEngagement(of: kind, in: [gliaConfiguration.queueId])
         } catch {
             print("Error starting engagement")
         }
     }
+    
+    // Authenticate user with token to start secure conversation
+    // and display user data on operator side
+    func authenticate() {
+        gliaAuthentication?.authenticate(
+            with: gliaConfiguration.directIdToken,
+            accessToken: nil
+        ) { [weak self] result in
+            switch result {
+            case .success:
+                self?.isAuthenticated = true
+            case .failure(let error):
+                print("Error has happened: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    func deauthenticate() {
+        gliaAuthentication?.deauthenticate { [weak self] result in
+            switch result {
+            case .success:
+                self?.isAuthenticated = false
+            case .failure(let error):
+                print("Error has happened: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    // Show the visitor code to display code and start engagements with live observation
+    func showVisitorCode() {
+        var topViewController: UIViewController?
+        for scene in UIApplication.shared.connectedScenes {
+            if let windowScene = scene as? UIWindowScene {
+                for window in windowScene.windows {
```

Comment:
This can take advantage of `where`: `for window in windowScene.windows where window.isKeyWindow` to remove inner `if`. 

I wonder if we need to also break from the loop when key window is found to avoid more iterations.
Also it seems possible to have several key windows at once, that means `topViewController` could be assigned several times. But finding first one key window and break from loop is likely enough for now.

Reply:
Agree. 

Commit https://github.com/salemove/ios-sdk-widgets/pull/1049/commits/a103b7aba199f55f823e5eb20fe7f39ce13c2feb